### PR TITLE
fix: Upgrade dependencies and fix unit tests

### PR DESCRIPTION
**Pull request by raulkele**
fix: Upgrade dependencies and fix unit tests

Changelog:

- upgraded vulnerable dependencies (@rchincha)
- upgraded eslint version and config setup from .eslintrc to the new flat config file eslint.config.json
- upgraded unit test libraries
- updated unit tests failing due to breaking changes in jest, always use queryBy selectors when looking for negative result
- updated unit tests failing due to non awaited userEvent and fireEvent instances

No user facing changes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by rchincha** [comment]
```
Traceback (most recent call last):
  File "/home/runner/work/zui/zui/./tests/scripts/load_test_data.py", line 179, in <module>
    main()
  File "/home/runner/work/zui/zui/./tests/scripts/load_test_data.py", line 168, in main
    image_metadata = pull_modify_push_image(logger, registry, image_name, tag, cosign_password, multiarch, username, password, debug, data_dir)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/zui/zui/./tests/scripts/load_test_data.py", line 91, in pull_modify_push_image
    image_metadata["cves"] = getCVEInfo(image_metadata.pop("trivy"))
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^
KeyError: 'trivy'
make: *** [Makefile:30: test-data] Error 1
```

**Comment by rchincha** [comment]
https://productionresultssa14.blob.core.windows.net/actions-results/27d9ceda-7dde-47d2-8d13-4b67e31da8d1/workflow-job-run-be3eeb22-584a-5ee4-236b-1a042e19f0eb/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-12-24T21%3A42%3A51Z&sig=BvxV0zXAntDQ%2BlbOK4SFy6R7iCP2dxXF%2Bjwn3zkjF0g%3D&ske=2024-12-25T07%3A01%3A52Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2024-12-24T19%3A01%3A52Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2024-11-04&sp=r&spr=https&sr=b&st=2024-12-24T21%3A32%3A46Z&sv=2024-11-04

^ compared to this run, this PR is different. There is no 'trivy' field in the map.

**Comment by andaaron** [comment]
Fixed in https://github.com/project-zot/zui/pull/467, please rebase.

File: src/__tests__/Explore/Explore.test.js [hunk 1]
```diff
@@ -3,15 +3,15 @@ import userEvent from '@testing-library/user-event';
...
+import { createSearchParams, MemoryRouter } from 'react-router';
 import filterConstants from 'utilities/filterConstants.js';
 import { sortByCriteria } from 'utilities/sortCriteria.js';
 import MockThemeProvider from '__mocks__/MockThemeProvider';
 
 // router mock
 const mockedUsedNavigate = jest.fn();
-jest.mock('react-router-dom', () => ({
-  ...jest.requireActual('react-router-dom'),
+jest.mock('react-router', () => ({
```
Comment (rchincha) [inline src/__tests__/Explore/Explore.test.js:13]: This appears to be the repeated pattern ...

**Review by rchincha — APPROVED** [review]
lgtm

Thanks a lot @raulkele 

@andaaron pls take a look.

If it looks good, let's merge.

**Review by andaaron — APPROVED** [review]

**Comment by andaaron** [comment]
> lgtm
> 
> Thanks a lot @raulkele
> 
> @andaaron pls take a look.
> 
> If it looks good, let's merge.

We can merge from my point of view

**Pull request merged**